app/controllers/comment_controller.rb

```diff
@@ -44,7 +44,7 @@ def create_by_token
     @node = Node.find params[:id]
     @user = User.find_by_username params[:username]
     @body = params[:body]
-    @token = request.env['rack.session']['token']
+    @token = request.headers["HTTP_TOKEN"]
```

Comment:
@ryzokuken Can you check this with `@request.headers["HTTP_TOKEN"]` and check if the tests pass.